File: java_backend/src/main/java/com/example/dto/RoadMap.java
package com.example.model;

import java.util.Date;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class RoadMap {
	private String id;
	private String user_id;
	private String title;
	private String objective;
	private String profile;
	private Date deadline;
	private Date created_at;
	private Date updated_at;
}

File: flutter_client/lib/common/model/road_map.dart
import 'package:flutter_client/common/model/node.dart';
import 'package:freezed_annotation/freezed_annotation.dart';

part '_generated/road_map.freezed.dart';
part '_generated/road_map.g.dart';

@freezed
class RoadMap with _$RoadMap {
  const RoadMap._();
  const factory RoadMap({
    required String id,
    required String title,
    required String objective,
    required String profile,
    required DateTime deadline,
    @JsonKey(name: 'created_at') required DateTime createdAt,
    @JsonKey(name: 'updated_at') required DateTime updatedAt,
    @Default({}) Map<String, Node> nodes,
  }) = _RoadMap;

  factory RoadMap.fromJson(Map<String, dynamic> json) =>
      _$RoadMapFromJson(json);

  RoadMap setNodes(Map<String, Node> newNodes) {
    return copyWith(nodes: newNodes);
  }
}


Map<String, Node> listToNodeMap(List<Node> nodeList) {
  return {for (var node in nodeList) node.id: node};
}
